src/extra/imgui.cpp

```diff
@@ -1247,323 +1392,2342 @@ struct TreeNode : public Base {
+                               void *data, int count, double width, int offset);
+
+  double _width = 0.67;
+  bool _horizontal = false;
+  PlotBarsProc _plot = &ImPlot::PlotBarsG;
+
+  static inline Parameters params{
+      PlottableBase::params,
+      {{"Width", CBCCSTR("The width of each bar"), {CoreInfo::FloatType}},
+       {"Horizontal",
+        CBCCSTR("If the bar should be horiziontal rather than vertical"),
+        {CoreInfo::BoolType}}},
+  };
+
+  static CBParametersInfo parameters() { return params; }
+
+  void setParam(int index, const CBVar &value) {
+    switch (index) {
+    case PlottableBase::nparams + 1:
+      _width = value.payload.floatValue;
+      break;
+    case PlottableBase::nparams + 2:
+      _horizontal = value.payload.boolValue;
+      if (_horizontal) {
+        _plot = &ImPlot::PlotBarsHG;
+      } else {
+        _plot = &ImPlot::PlotBarsG;
+      }
+      break;
+    default:
+      PlottableBase::setParam(index, value);
+    }
+  }
+
+  CBVar getParam(int index) {
+    switch (index) {
+    case PlottableBase::nparams + 1:
+      return Var(_width);
+    case PlottableBase::nparams + 2:
+      return Var(_horizontal);
+      break;
+    default:
+      return PlottableBase::getParam(index);
+    }
+  }
+
+  CBVar activate(CBContext *context, const CBVar &input) {
+    if (!isInputValid(input))
+      return input;
+
+    PlottableBase::applyModifiers();
+    DEFER(PlottableBase::popModifiers());
+    if (_kind == Kind::xAndY) {
+      _plot(
+          _fullLabel.c_str(),
+          [](void *data, int idx) {
+            auto input = reinterpret_cast<CBVar *>(data);
+            auto seq = input->payload.seqValue;
+            return ImPlotPoint(seq.elements[idx].payload.float2Value[0],
+                               seq.elements[idx].payload.float2Value[1]);
+          },
+          (void *)&input, int(input.payload.seqValue.len), _width, 0);
+    } else if (_kind == Kind::xAndIndex) {
+      _plot(
+          _fullLabel.c_str(),
+          [](void *data, int idx) {
+            auto input = reinterpret_cast<CBVar *>(data);
+            auto seq = input->payload.seqValue;
+            return ImPlotPoint(double(idx),
+                               seq.elements[idx].payload.floatValue);
+          },
+          (void *)&input, int(input.payload.seqValue.len), _width, 0);
+    }
+    return input;
+  }
+};
+
+struct HasPointer : public Base {
+  static CBTypesInfo inputTypes() { return CoreInfo::NoneType; }
+  static CBTypesInfo outputTypes() { return CoreInfo::BoolType; }
+  static CBVar activate(CBContext *context, const CBVar &input) {
+    return Var(::ImGui::IsAnyItemHovered());
+  }
+};
+
+struct FPS : public Base {
+  static CBTypesInfo outputTypes() { return CoreInfo::FloatType; }
+
+  CBVar activate(CBContext *context, const CBVar &input) {
+    ImGuiIO &io = ::ImGui::GetIO();
+    return Var(io.Framerate);
+  }
+};
+
+struct Version : public Base {
```

Comment:
We don't need this.
In fact `GUI` could be replaced one day with another library, we should detach from `ImGui`